### PR TITLE
feat: download button

### PR DESCRIPTION
**Pull request by DanChae03**
# Description

Reusable download button for the videos watch page

# How should this PR be QA Tested?

- [ ] Should render correct button/icon based on whether prop passed in is 'button' or 'icon'
- [ ] Should open download dialog on press

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged into main

### CODE REVIEW

**Review by GeronimoJohn — APPROVED** [review]

File: apps/watch/src/components/VideoContainerPage/DownloadButton/DownloadButton.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { ReactElement } from 'react'
```
Comment (storyworks) [inline apps/watch/src/components/VideoContainerPage/DownloadButton/DownloadButton.tsx:1]: Watch where you put this component. Download is not used in the container page at all, only in the content page

Can you move Download button to the VideoContentPage folder and in this PR also move ShareButton out of this VideoContainer folder and into the main components folder?

**Review by storyworks — CHANGES_REQUESTED** [review]

**Review by storyworks — APPROVED** [review]

File: apps/watch/src/components/VideoContentPage/DownloadButton/DownloadButton.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { ReactElement } from 'react'
+import Button from '@mui/material/Button'
+import FileDownloadOutlinedIcon from '@mui/icons-material/FileDownloadOutlined'
+import IconButton from '@mui/material/IconButton'
+
+export interface DownloadButtonProps {
+  variant: 'button' | 'icon'
+  openDialog: () => void
```
Comment (storyworks) [inline apps/watch/src/components/VideoContentPage/DownloadButton/DownloadButton.tsx:8]: rename this `onClick` 

Download button onClick should not be linked to a name of an action - in the future we may choose to add event tracking onClick as well, or redirect to a share page.

Same with share button if you've used that there

File: apps/watch/src/components/VideoContentPage/DownloadButton/DownloadButton.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  })
+
+  it('should render icon button based on variant correctly', () => {
+    const { getByTestId } = render(
+      <DownloadButton variant="icon" openDialog={noop} />
+    )
+    expect(getByTestId('FileDownloadOutlinedIcon')).toBeInTheDocument()
+  })
+
+  it('should execute download button operation', () => {
```
Comment (storyworks) [inline apps/watch/src/components/VideoContentPage/DownloadButton/DownloadButton.spec.tsx:20]: `it should call onClick when button is clicked`

**Review by storyworks — CHANGES_REQUESTED** [review]

File: apps/watch/src/components/VideoContentPage/DownloadButton/DownloadButton.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { render, fireEvent } from '@testing-library/react'
+import { noop } from 'lodash'
+import { DownloadButton } from './DownloadButton'
+
+describe('DownloadButton', () => {
+  it('should render download button based on variant correctly', () => {
```
Comment (storyworks) [inline apps/watch/src/components/VideoContentPage/DownloadButton/DownloadButton.spec.tsx:6]: `it should render button variant as button` and `it should render icon variant as icon`?

File: apps/watch/src/components/VideoContentPage/DownloadButton/DownloadButton.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      onClick={openDialog}
+      sx={{
+        display: { xs: 'none', sm: 'flex' },
+        minWidth: '200px'
+      }}
+    >
+      Download
+    </Button>
+  ) : (
+    <IconButton onClick={openDialog}>
```
Comment (storyworks) [inline apps/watch/src/components/VideoContentPage/DownloadButton/DownloadButton.tsx:30]: can you add `data-testid="download"` to this so we can test it?

File: apps/watch/src/components/VideoContentPage/DownloadButton/DownloadButton.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  it('should execute download button operation', () => {
+    const setOpenDownload = jest.fn()
+
+    const { getByRole } = render(
+      <DownloadButton variant="button" openDialog={setOpenDownload} />
+    )
+
+    fireEvent.click(getByRole('button'))
+    expect(setOpenDownload).toHaveBeenCalled()
+  })
```
Comment (storyworks) [inline apps/watch/src/components/VideoContentPage/DownloadButton/DownloadButton.spec.tsx:29]: Also add another test `it should call onClick when icon button is clicked`. Any changes to the onClick with icon button isn't picked up right now.

Make this change also for Share

**Review by storyworks — CHANGES_REQUESTED** [review]

File: apps/watch/src/components/ShareButton/ShareButton.tsx [hunk 7]
```diff
@@ -5,29 +5,30 @@ import IconButton from '@mui/material/IconButton'
...
     <Button
       variant="outlined"
       startIcon={<ShareOutlinedIcon />}
       size="medium"
       color="secondary"
-      onClick={openDialog}
+      onClick={onClick}
+      data-testid="share"
       sx={{
         display: { xs: 'none', sm: 'flex' },
```
Comment (storyworks) [inline apps/watch/src/components/ShareButton/ShareButton.tsx:24]: Remove this logic - ShareButton itself should not care how and when it's displayed. That's the role of the parent components.

**Review by storyworks — APPROVED** [review]

**Pull request merged**